source/errors.go

```diff
@@ -1,15 +1,18 @@
 package source
 
-import "os"
-
 // ErrDuplicateMigration is an error type for reporting duplicate migration
 // files.
 type ErrDuplicateMigration struct {
 	Migration
-	os.FileInfo
+	FileInfo
```

Comment:
Why not use `os.FileInfo` or [fs.FileInfo](https://tip.golang.org/pkg/io/fs/#FileInfo)?

You can use [fs.DirEntry.Info()](https://tip.golang.org/pkg/io/fs/#DirEntry) to get the `fs.FileInfo`